src/lib/templateResource.ts

```diff
@@ -0,0 +1,40 @@
+type ResourceReducer =
+  | ((resource: string, config: Config) => string)
+  | ((resource: string) => string);
+
+// Removes all spaces from string, this allows the other reducers to have more
+// reliable match patterns as they don't need to be concerned about whitespace.
+function removeWhitespace(resource: string): string {
+  return resource.replace(/\s/g, "");
+}
+
+// Replaces the template placeholder <% RANDOM %> with a random string
+function replaceRandom(resource: string): string {
+  const dictionary =
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+  const dictionaryLength = dictionary.length;
+  let random = "";
+  for (let i = 0; i < 16; i++) {
+    random += dictionary.charAt(Math.floor(Math.random() * dictionaryLength));
+  }
+  return resource.replace("<%RANDOM%>", random);
```

Comment:
Do you anticipate needing to replace multiple `<%RANDOM%>`? It seems as if currently it will only replace 1.